pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerBase.java

```diff
@@ -193,6 +194,11 @@ protected ConsumerBase(PulsarClientImpl client, String topic, ConsumerConfigurat
         initReceiverQueueSize();
     }
 
+    @VisibleForTesting
+    public UnAckedMessageTracker getUnAckedMessageTracker() {
+        return unAckedMessageTracker;
+    }
+
```

Comment:
There is no need to make it public only for tests. You can remove the `public` here and add back the `getUnAckedMessageTracker` to `ConsumerImpl`.

Reply:
Both `ConsumerImpl` and `MultiTopicsConsumerImpl` have `getUnAckedMessageTracker`,  it's better to move it to `ConsumerBase`.
The `VisibleForTesting` annotation should not be added here, will remove it.